pkg/platform/types.go

```diff
@@ -181,8 +181,38 @@ type UpdateProjectOptions struct {
 	ProjectConfig ProjectConfig
 }
 
+type DeleteProjectStrategy string
+
+const (
+
+	// DeleteProjectStrategyCascade - delete sub resources along with project leave no orphans behind
+	DeleteProjectStrategyCascade DeleteProjectStrategy = "cascade"
+
+	// DeleteProjectStrategyRestrict - avoid deleting when project contains sub resources (e.g.: functions)
+	DeleteProjectStrategyRestrict DeleteProjectStrategy = "restrict"
+
+	// TODO: add mode to delete sub resources prior to project deletion
```

Comment:
Why this is not the behavior in `cascade` mode ?

Reply:
Actually, no real good reason. will do